Depricated GetDataOnSessionKey() cannot be overloaded in cross-compiling

### Bug Report
When building with some toolchain, all the new type of GetDataOnSessionKey() functions newly added throughout some classes are conflicting with old type of same function that has been declared as deprecated in 4.5.0

##### Reproduction Steps
1. Download latest version (4.5.0) from master branch
2. Build with ARM toolchain (like 'arm-none-linux-gnueabi')
3. And wait for compilation errors related to the overload

##### Expected Behavior
It should be successful even if it was built by other toolchains instead of x86_64

##### Observed Behavior
The compiler emitted errors against all functions of GetDataOnSessionKey().
It says that two different type of this function can not be overloaded

`In file included from /home/bhkwon/sdl/core/src/components/include/connection_handler/connection_handler.h:38:0,
                 from /home/bhkwon/sdl/core/src/components/connection_handler/src/connection.cc:38:
/home/bhkwon/sdl/core/src/components/include/protocol_handler/session_observer.h:255:19: error: 'virtual int32_t protocol_handler::SessionObserver::GetDataOnSessionKey(uint32_t, uint32_t*, std::list<int>*, transport_manager::DeviceHandle*) const' cannot be overloaded
   virtual int32_t GetDataOnSessionKey(
                   ^
/home/bhkwon/sdl/core/src/components/include/protocol_handler/session_observer.h:242:19: error: with 'virtual int32_t protocol_handler::SessionObserver::GetDataOnSessionKey(uint32_t, uint32_t*, std::list<int>*, uint32_t*) const'
   virtual int32_t GetDataOnSessionKey(uint32_t key,
                   ^
In file included from /home/bhkwon/sdl/core/src/components/connection_handler/src/connection.cc:38:0:
/home/bhkwon/sdl/core/src/components/include/connection_handler/connection_handler.h:194:30: error: 'virtual int32_t connection_handler::ConnectionHandler::GetDataOnSessionKey(uint32_t, uint32_t*, std::list<int>*, uint32_t*) const' cannot be overloaded
   DEPRECATED virtual int32_t GetDataOnSessionKey(`


##### OS & Version Information
* OS/Version: Ubuntu 14.04 LTS (build machine)
* SDL Core Version: 4.5.0
* Testing Against: NA

--- a/src/components/connection_handler/include/connection_handler/connection_handler_impl.h
+++ b/src/components/connection_handler/include/connection_handler/connection_handler_impl.h
@@ -503,20 +503,6 @@ class ConnectionHandlerImpl
       std::list<int32_t>* sessions_list,
       connection_handler::DeviceHandle* device_id) const OVERRIDE;
 
-  /**
-   * DEPRECATED
-   * \brief information about given Connection Key.
-   * \param key Unique key used by other components as session identifier
-   * \param app_id Returned: ApplicationID
-   * \param sessions_list Returned: List of session keys
-   * \param device_id Returned: DeviceID
-   * \return int32_t -1 in case of error or 0 in case of success
-   */
-  int32_t GetDataOnSessionKey(uint32_t key,
-                              uint32_t* app_id,
-                              std::list<int32_t>* sessions_list,
-                              uint32_t* device_id) const OVERRIDE;
-
   const ConnectionHandlerSettings& get_settings() const OVERRIDE;
 
   const protocol_handler::SessionObserver& get_session_observer();

--- a/src/components/connection_handler/src/connection_handler_impl.cc
+++ b/src/components/connection_handler/src/connection_handler_impl.cc
@@ -709,18 +709,6 @@ int32_t ConnectionHandlerImpl::GetDataOnSessionKey(
   return 0;
 }
 
-int32_t ConnectionHandlerImpl::GetDataOnSessionKey(
-    uint32_t key,
-    uint32_t* app_id,
-    std::list<int32_t>* sessions_list,
-    uint32_t* device_id) const {
-  LOG4CXX_AUTO_TRACE(logger_);
-  DeviceHandle handle;
-  int32_t result = GetDataOnSessionKey(key, app_id, sessions_list, &handle);
-  *device_id = static_cast<uint32_t>(handle);
-  return result;
-}
-
 const ConnectionHandlerSettings& ConnectionHandlerImpl::get_settings() const {
   return settings_;
 }

--- a/src/components/include/connection_handler/connection_handler.h
+++ b/src/components/include/connection_handler/connection_handler.h
@@ -183,21 +183,6 @@ class ConnectionHandler {
       connection_handler::DeviceHandle* device_id) const = 0;
 
   /**
-   * DEPRECATED
-   * \brief information about given Connection Key.
-   * \param key Unique key used by other components as session identifier
-   * \param app_id Returned: ApplicationID
-   * \param sessions_list Returned: List of session keys
-   * \param device_id Returned: DeviceID
-   * \return int32_t -1 in case of error or 0 in case of success
-   */
-  DEPRECATED virtual int32_t GetDataOnSessionKey(
-      uint32_t key,
-      uint32_t* app_id,
-      std::list<int32_t>* sessions_list,
-      uint32_t* device_id) const = 0;
-
-  /**
    * @brief GetConnectedDevicesMAC allows to obtain MAC adresses for all
    * currently connected devices.
    *

--- a/src/components/include/protocol_handler/session_observer.h
+++ b/src/components/include/protocol_handler/session_observer.h
@@ -228,20 +228,6 @@ class SessionObserver {
                            uint8_t* sessionId) const = 0;
 
   /**
-   * DEPRECATED
-   * \brief information about given Connection Key.
-   * \param key Unique key used by other components as session identifier
-   * \param app_id Returned: ApplicationID
-   * \param sessions_list Returned: List of session keys
-   * \param device_id Returned: DeviceID
-   * \return int32_t -1 in case of error or 0 in case of success
-   */
-  virtual int32_t GetDataOnSessionKey(uint32_t key,
-                                      uint32_t* app_id,
-                                      std::list<int32_t>* sessions_list,
-                                      uint32_t* device_id) const = 0;
-
-  /**
    * \brief information about given Connection Key.
    * \param key Unique key used by other components as session identifier
    * \param app_id Returned: ApplicationID

